### ISSUE TITLE
This application is modifying the autolayout engine from a background thread...

### ISSUE THREAD

**Issue by screenworker** [fulldecent/FDWaveformView#73]
I tried the your example (XCode 8.1):

```
//Inside UIStackView (Autolayout)
 @IBOutlet weak var waveformView: FDWaveformView!

override func viewDidLoad() {
            let thisBundle = Bundle.main
            let url = thisBundle.url(forResource: "bass2", withExtension: "caf")
            self.waveformView.audioURL = url
}
```
... and got:

> This application is modifying the autolayout engine from a background thread, which can lead to engine corruption and weird crashes. This will cause an exception in a future release."

**Stack Trace:**

This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   CoreFoundation                      0x0000000189ef01d8 <redacted> + 148
	1   libobjc.A.dylib                     0x000000018892855c objc_exception_throw + 56
	2   CoreFoundation                      0x0000000189ef0108 <redacted> + 0
	3   Foundation                          0x000000018aad4f0c <redacted> + 192
	4   Foundation                          0x000000018aad4c4c <redacted> + 76
	5   Foundation                          0x000000018a9205dc <redacted> + 112
	6   Foundation                          0x000000018aad38e8 <redacted> + 112
	7   UIKit                               0x000000018fd368e4 <redacted> + 220
	8   FDWaveformView                      0x000000010014f4dc _TFC14FDWaveformView14FDWaveformView14layoutSubviewsfT_T_ + 80
	9   FDWaveformView                      0x0000000100150820 _TToFC14FDWaveformView14FDWaveformView14layoutSubviewsfT_T_ + 40
	10  UIKit                               0x000000018fd3654c <redacted> + 1196
	11  QuartzCore                          0x000000018d1fe40c <redacted> +
(0.0, 259.0, 375.0, 129.0) -- (0.0, 0.0, 374.999119685246, 129.0)
2016-11-20 00:11:10.778657 xxxxxxxxx[30793:7181186] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   CoreFoundation                      0x0000000189ef01d8 <redacted> + 148
	1   libobjc.A.dylib                     0x000000018892855c objc_exception_throw + 56
	2   CoreFoundation                      0x0000000189ef0108 <redacted> + 0
	3   Foundation                          0x000000018aad4f0c <redacted> + 192
	4   Foundation                          0x000000018aad4c4c <redacted> + 76
	5   Foundation                          0x000000018a9205dc <redacted> + 112
	6   Foundation                          0x000000018aad38e8 <redacted> + 112
	7   UIKit                               0x000000019003aa48 <redacted> + 172
	8   UIKit                               0x000000018fd36584 <redacted> + 1252
	9   QuartzCore                          0x000000018d1fe40c <redacted> + 148
	10  QuartzCore                          0x000000018d1f30e8 <redacted> + 292
	11  QuartzCore                          0x000000018d1f2fa8 <redacted> + 32
	12  QuartzCore                          0x000000018d16fc64 <redacted> + 252
	13  QuartzCor
2016-11-20 00:11:10.784722 xxxxxxxxx[30793:7181186] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   CoreFoundation                      0x0000000189ef01d8 <redacted> + 148
	1   libobjc.A.dylib                     0x000000018892855c objc_exception_throw + 56
	2   CoreFoundation                      0x0000000189ef0108 <redacted> + 0
	3   Foundation                          0x000000018aad4f0c <redacted> + 192
	4   Foundation                          0x000000018aad4c4c <redacted> + 76
	5   Foundation                          0x000000018a9205dc <redacted> + 112
	6   Foundation                          0x000000018aad38e8 <redacted> + 112
	7   UIKit                               0x000000018fd36738 <redacted> + 1688
	8   QuartzCore                          0x000000018d1fe40c <redacted> + 148
	9   QuartzCore                          0x000000018d1f30e8 <redacted> + 292
	10  QuartzCore                          0x000000018d1f2fa8 <redacted> + 32
	11  QuartzCore                          0x000000018d16fc64 <redacted> + 252
	12  QuartzCore                          0x000000018d1970d0 <redacted> + 512
	13  QuartzCor

**Comment by vatsal1992** [comment]
**Xcode Version 8.1 (8B62), Swift3, iOS 10.1.1**
i got same issue as @screenworker  and when i play using audio it starts with some progress at right side. (Picture Shown, iPhone6s) but it works good in iPhone 5s.
![img_6090](https://cloud.githubusercontent.com/assets/12713108/20552130/c60935e4-b16d-11e6-8c3a-cab3a0862fb0.JPG) <br>
here is my code to set progress<br>
```
if let total = weakSelf?.myPlayer.duration {
                            let percentage = Float(current / total)
                            let endSamples = cell.waveView.zoomEndSamples
                            let startSamples = cell.waveView.zoomStartSamples
                            let rangeSamples = Float(endSamples - startSamples)
                            let location = percentage
                            let progress = Int(Float(startSamples) + rangeSamples * location)
                            let final = progress
                            let dFormat = "%02d"
                            let min:Int = Int(current / 60)
                            let sec:Int = Int(current.truncatingRemainder(dividingBy: 60))
                            let remaining:Int = Int(total) - sec
                            let s = "\(String(format: dFormat, min)):\(String(format: dFormat, remaining))"
                            DispatchQueue.main.async(execute: {
                                cell.waveView.progressSamples = Int(final)
                                cell.lblDuration.text = s
                            })
                        }
```

**Comment by fulldecent** [comment]
Fixed in #75, thanks @steryokhin
